### PR TITLE
Introduce support for more functions to the AST. Simplify function mappings in the Parser

### PR DESCRIPTION
**Pull request by izeigerman**

### CODE REVIEW

File: sqlglot/expressions.py [hunk 1]
```diff
@@ -512,23 +514,39 @@ def from_arg_list(cls, args):
...
-        return cls._sql_name
+        return cls.sql_names()[0]
 
     @classmethod
     def default_parser_mappings(cls):
-        return {cls.sql_name: cls.from_arg_list}
+        return {name: cls.from_arg_list for name in cls.sql_names()}
+
+
+class AggFunc(Func):
```
Comment (tobymao) [inline sqlglot/expressions.py:536]: maybe just agg ?

**Comment by tobymao** [comment]
looks great, while were here can you add the common stats ones like stddev, variance, covariance etc and test them in identity.sql

**Comment by izeigerman** [comment]
@tobymao great suggestion! Will do both

File: sqlglot/expressions.py (hunk 1 above)
Comment (izeigerman) [reply sqlglot/expressions.py:536]: I don't mind but I'm a little concerned that in this case there is no way to tell that's this is actually a base class and not an actual function.

Comment (tobymao) [reply sqlglot/expressions.py:536]: should we use ABC?

Comment (tobymao) [reply sqlglot/expressions.py:536]: i'm fine to leave it aggfunc and no ABC, just a thought

Comment (izeigerman) [reply sqlglot/expressions.py:536]: Got it, thanks. If your opinion on this is not super strong, I'd prefer to keep it `AggFunc` to make it stand out from concrete functions.

**Comment by tobymao** [comment]
@izeigerman one last test in expression.py to make sure it's actually getting parsed as an exp.X

**Review by tobymao — APPROVED** [review]

**Comment by izeigerman** [comment]
@tobymao added stddev and variance. I haven't added correlation and covariance because I noticed that it's not supported by MySQL out of the box.

**Pull request merged**